### PR TITLE
[CI] Splitting workflows for PR/Push (short vs full) CI runs on 2.4.6

### PR DESCRIPTION
**Pull request by scunniff**
nomad-spark PR Summary
==============================
See https://github.com/ExpediaGroup/nomad-spark/pull/30. This is the equivalent PR for the 2.4.6 branch, where the main differences are:

- no longer need to alter the run-tests script as it is not used in CI for 2.4.6 (sql/hive can be built like any other module)
- the branch-2.4 workflow was deleted as it is not used
- hadoop-cloud is not supported for 2.4.6 (runs into issues immediately), so not included in build

Additional Details
==============================

See https://github.com/scunniff/nomad-spark/runs/1641655252 for completed runs of both workflows.

Checklist
==============================

Testing
-------------------------

(Remove this checklist and replace it with "N/A - no code changes" if this PR does not modify source code)

* [x] I have manually verified that my code changes do the right thing.
* [x] I have run all tests and verified that my changes do not introduce any regressions.
* [ ] I have written unit tests to verify that my code changes do the right thing and to protect my code against regressions.
* [ ] I have tested my code changes locally or against a cluster using an example Spark project.

Documentation
-------------------------

N/A - no code changes

### CODE REVIEW

File: .github/workflows/mvn-branch-ci.yaml [hunk 1]
```diff
@@ -1,9 +1,6 @@
-name: Build Branch
+name: Build Branch (PR)
```
Comment (pthatchinamoorthy) [inline .github/workflows/mvn-branch-ci.yaml:1]: How would this be triggered ?
On raising a PR, should the Developer manually trigger the this CI Action ?

File: .github/workflows/mvn-branch-full-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+name: Build Branch (Full)
```
Comment (pthatchinamoorthy) [inline .github/workflows/mvn-branch-full-ci.yaml:1]: Technically, when you add an commit to the feature branch, the build would be triggered, correct ?

File: .github/workflows/mvn-branch-ci.yaml (hunk 1 above)
Comment (pthatchinamoorthy) [reply .github/workflows/mvn-branch-ci.yaml:1]: ahh.. i see 

```
pull_request:
    branches:
      "**"	
```

**Review by pthatchinamoorthy — APPROVED** [review]

**Review by pthatchinamoorthy — CHANGES_REQUESTED** [review]
I just read the other PR for `3.0.1`

So, the method `identify_changed_files_from_git_commits` in `run-tests.py` cleverly picks the modules that got changed, and runs the tests only for those modules that got updated -  which is our objective as well.

I see that in branch `2.4.6`, the above logic gets implemented only for `test_env == "amplab_jenkins"`. Do you think, we just have to leverage that mechanishm and update the `run-tests.py` to run in our jenkins as well, and update our github workflow accordingly.

File: .github/workflows/mvn-branch-ci.yaml (hunk 1 above)
Comment (scunniff) [reply .github/workflows/mvn-branch-ci.yaml:1]: This is triggered solely from PRs automatically.

**Review by pthatchinamoorthy — APPROVED** [review]

**Pull request merged**